local_pathfinding/objectives.py

```diff
@@ -24,66 +53,313 @@ def motionCost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
+
+        return ob.Cost(distance_m)
 
 
 class MinimumTurningObjective(Objective):
-    def __init__(self, space_information, simple_setup, heading_degrees: float):
+    """Generates a minimum turning objective function
+
+    Attributes:
+        goal_x (float): The x coordinate of the goal state
+        goal_y (float): The y coordinate of the goal state
+        heading (float): The heading of the sailbot in radians (-pi, pi]
+        method (MinimumTurningMethod): The method of the minimum turning objective function
+    """
+
+    def __init__(
+        self,
+        space_information,
+        simple_setup,
+        heading_degrees,
+        method: MinimumTurningMethod,
+    ):
         super().__init__(space_information)
-        self.simple_setup = simple_setup
-        self.heading_degrees = heading_degrees
+        self.goal_x = simple_setup.getGoal().getState().getX()
+        self.goal_y = simple_setup.getGoal().getState().getY()
+        self.heading = math.radians(heading_degrees)
+        self.method = method
 
     def motionCost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
-        return ob.Cost()
+        """Generates the turning cost between s1, s2, heading or the goal position
 
-    def _goal_path_turn_cost(self, s1, s2):
-        raise NotImplementedError
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
 
-    def _goal_heading_turn_cost(self, s1):
-        raise NotImplementedError
+        Returns:
+            ob.Cost: The minimum turning angle (degrees)
 
-    def _heading_path_turn_cost(self, s1, s2):
-        raise NotImplementedError
+        Raises:
+            ValueError: If the minimum turning method is not supported
+        """
+
+        if self.method == MinimumTurningMethod.GOAL_HEADING:
+            # Calculate the minimum turning cost between s1-goal and heading
+            return self.goal_heading_turn_cost(s1)
+        elif self.method == MinimumTurningMethod.GOAL_PATH:
+            # Calculate the minimum turning cost between s1-s2 and s1-goal
+            return self.goal_path_turn_cost(s1, s2)
+        elif self.method == MinimumTurningMethod.HEADING_PATH:
+            # Calculate the minimum turning cost from sl-s2 and heading
+            return self.heading_path_turn_cost(s1, s2)
+        else:
+            ValueError(f"Method {self.method} not supported")
+
+    def goal_path_turn_cost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
+        """Determine the smallest turn angle between s1-s2 and s1-goal
+
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
+
+        Returns:
+            ob.Cost: the turning angle from s2 to s1 (degrees)
+        """
+
+        # Calculate the angle of the s1-s2 line segment from North
```

Comment:
more specifically, the comment could say `# Calculate the true bearing of s2 from s1`